### PR TITLE
Invalidate Session Configuration

### PR DESCRIPTION
**Pull request by ricardopereira**
- A way for the user customize the session configuration using the delegate (i.e. change the `allowsCellularAccess` flag).

### CODE REVIEW

**Comment by Heikowi** [comment]
Thank you for your contribution!
There is still a warning left:
```
HWIFileDownloader.m:1485:15: warning: unused variable 'oldSessionIdentifier' [-Wunused-variable]
    NSString *oldSessionIdentifier = self.backgroundSessionIdentifier;
              ^
1 warning generated.
```

**Comment by ricardopereira** [comment]
@Heikowi Ups, thanks for reviewing. I'll fix it.

**Comment by ricardopereira** [comment]
@Heikowi Updated! Have a look please.
I updated the sample code with a little demonstration of what I implemented and how it could be used.

**Comment by Heikowi** [comment]
Thanks for the demo code!

The button action method can be more condensed. What do you think?

```
- (void)useMobileDataSwitched:(UISwitch *)useMobileDataSwitch {
    DemoDownloadAppDelegate *theAppDelegate = (DemoDownloadAppDelegate *)[UIApplication sharedApplication].delegate;
    theAppDelegate.demoDownloadStore.allowsCellularAccess = useMobileDataSwitch.isOn;

    BOOL isReachableViaWiFi = NO; // SCNetworkReachability
    
    BOOL cancelDownloadTasks = theAppDelegate.demoDownloadStore.allowsCellularAccess ? NO : !isReachableViaWiFi;
    
    [theAppDelegate.fileDownloader invalidateSessionConfigurationAndCancelTasks:cancelDownloadTasks];

    // after cancelDownloadTasks -> restart all downloads
}
```

File: Demo/HWIFileDownload/DemoDownloadStore.m [hunk 1]
```diff
@@ -395,17 +395,10 @@ - (nullable NSProgress *)rootProgress
...
 }
 
-/*
 - (void)customizeBackgroundSessionConfiguration:(nonnull NSURLSessionConfiguration *)aBackgroundSessionConfiguration
 {
-    NSMutableDictionary *aHTTPAdditionalHeadersDict = [aBackgroundSessionConfiguration.HTTPAdditionalHeaders mutableCopy];
-    if (aHTTPAdditionalHeadersDict == nil) {
-        aHTTPAdditionalHeadersDict = [[NSMutableDictionary alloc] init];
-    }
-    [aHTTPAdditionalHeadersDict setObject:@"identity" forKey:@"Accept-Encoding"];
```
Comment (Heikowi) [inline Demo/HWIFileDownload/DemoDownloadStore.m]: You deleted these lines. It is not necessary anymore to send the "identity" value in the "Accept-Encoding" header? I remember adding these lines in the context of these discussions (https://stackoverflow.com/questions/36581495/how-to-get-nsurlsession-to-return-content-lengthhttp-header-from-server-objec).

File: Demo/HWIFileDownload.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -300,7 +322,7 @@
 					/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include,
 				);
 				INFOPLIST_FILE = HWIFileDownload/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 6.0;
```
Comment (Heikowi) [inline Demo/HWIFileDownload.xcodeproj/project.pbxproj]: Please try to keep `IPHONEOS_DEPLOYMENT_TARGET = 6.0`.

Comment (ricardopereira) [reply Demo/HWIFileDownload.xcodeproj/project.pbxproj]: @Heikowi I tried but Xcode 10 doesn't support 6 as deployment target. Even 7 or 8 isn't support anymore. Only iOS 9 up. 🤷‍♂️

File: Demo/HWIFileDownload/DemoDownloadStore.m (hunk 1 above)
Comment (ricardopereira) [reply Demo/HWIFileDownload/DemoDownloadStore.m]: @Heikowi Wasn't aware of that, sorry. I just saw a commented code and I thought it was some sort of old code that wasn't needed anymore. I will revert that.

**Comment by ricardopereira** [comment]
> The button action method can be more condensed. What do you think?

@Heikowi Agree 👍

File: Demo/HWIFileDownload/DemoDownloadStore.m (hunk 1 above)
Comment (ricardopereira) [reply Demo/HWIFileDownload/DemoDownloadStore.m]: Reverted 3ff0804.

File: Demo/HWIFileDownload.xcodeproj/project.pbxproj (hunk 2 above)
Comment (ricardopereira) [reply Demo/HWIFileDownload.xcodeproj/project.pbxproj]: Have in mind that this is only for the Demo. The library still works on older iOS versions.

**Comment by Heikowi** [comment]
Thanks!

**Pull request merged**